rewrite scfw preview doc (#37761)
DataDog/documentation: content/en/security/code_security/_index.md

Reviewer: ```suggestion
- [Supply Chain Security](#supply-chain-security-preview) for preventing malicious packages from entering your development environment and code repositories 
```
The renamed heading broke this link; this should fix it 🙂

Reviewer: ```suggestion
```
If you wind up using the callout I suggested, we don't need this anymore, since it doesn't use a reference-style link.

With [Cloud Security Management (CSM)][18], you can see misconfigurations in IaC Security directly from runtime findings. See [IaC Security setup][17] to get started.

## Supply Chain Security (Preview)

> Supply Chain Security prevents malicious open source packages from entering your development environments at the point of installation, before they reach your repositories or CI/CD pipelines. 

Author: done

> Unlike SCA, which scans dependencies already in your codebase, the Datadog Supply Chain Firewall (SCFW) intercepts package manager commands (`npm`, `pip`, `poetry`) in real time and blocks malicious, vulnerable, or recently published packages before they are installed.

Author: done

> Every package install is evaluated against Datadog's malicious package feed (powered by **Guarddog**), known vulnerability advisories, and configurable recency thresholds. When a package is flagged, installation is blocked immediately with a clear, actionable message(on both developer laptops and CI runners).

Author: no available link

> Every package install is evaluated against Datadog's malicious package feed (powered by **Guarddog**), known vulnerability advisories, and configurable recency thresholds. When a package is flagged, installation is blocked immediately with a clear, actionable message(on both developer laptops and CI runners).

Author: done

> As a Datadog product, SCFW goes beyond individual developer protection with:

Author: done

- Centralized policy management to define org-wide block lists, allow lists, and behavioral rules directly from the Datadog AppSec UI
- Event observability to search, filter, and audit ALLOW, WARN, and BLOCK events across developer machines and CI systems in a unified event feed

> - Coverage reporting to track which developers and repositories are protected, identify gaps, and measure your organization's supply chain security posture over time 

Author: done

> [Request access to the Preview][11].

Author: not yet have a setup page

With [Cloud Security Management (CSM)][18], you can see misconfigurations in IaC Security directly from runtime findings. See [IaC Security setup][17] to get started.

> ## Supply Chain Security (Preview)

Reviewer: Yeah yeah bot I fixed that (but thank you)

> ## Supply Chain Security (Preview)

Reviewer: ```suggestion
## Supply Chain Security (Preview)

{{< callout url=https://docs.google.com/forms/d/1Xqh5h1n3-jC7au2t30fdTq732dkTJqt_cb7C7T-AkPc/viewform?edit_requested=true
 btn_hidden="false" header="Join the Preview!">}}
Use this form to submit your request to join the Supply Chain Security Preview.
{{< /callout >}}
```
This is optional, but we have a standard callout for previews that we can add in here to make it a little more visible 🙂

Supply Chain Security prevents malicious open source packages from entering your development environments at the point of installation, before they reach your repositories or CI/CD pipelines.

Unlike SCA, which scans dependencies already in your codebase, the Datadog Supply Chain Firewall (SCFW) intercepts package manager commands (`npm`, `pip`, `poetry`) in real time and blocks malicious or recently published packages before they are installed.

> Every package install is evaluated against Datadog's malicious package feed (powered by **Guarddog**), known vulnerability advisories, and configurable recency thresholds. When a package is flagged, installation is blocked immediately with a clear, actionable message (on both developer laptops and CI runners).

Reviewer 2: nit: we usually spell it "GuardDog"

Author: done

> Every package install is evaluated against Datadog's malicious package feed (powered by **GuardDog**), known vulnerability advisories, and configurable recency thresholds. When a package is flagged, installation is blocked immediately with a clear, actionable message (on both developer laptops and CI runners).

Reviewer: ```suggestion
Supply Chain Security evaluates every package install against Datadog's malicious package feed (powered by GuardDog), known vulnerability advisories, and configurable recency thresholds. When it flags a package, it blocks installation immediately with a clear, actionable message (on both developer laptops and CI runners).
```
I wouldn't bold `GuardDog`, but if you wanted to, you could add a link to the [integration page](https://docs.datadoghq.com/integrations/guarddog/). I also replaced some passive voice with active here, so users can more easily understand what's doing what.

> In addition to protecting individual developer machines or CI pipelines, SCFW provides:
> 
> - Event observability to search, filter, and audit ALLOW, WARN, and BLOCK events across developer machines and CI systems in a unified event feed
> - *(coming soon)* Centralized policy management to define org-wide block lists, allow lists, and behavioral rules directly from the Datadog AppSec UI
> - *(coming soon)* Coverage reporting to track which developers and repositories are protected, identify gaps, and measure your organization's supply chain security posture over time

Reviewer: ```suggestion
In addition to protecting individual developer machines or CI pipelines, SCFW provides event observability to search, filter, and audit `ALLOW`, `WARN`, and `BLOCK` events across developer machines and CI systems in a unified event feed.
```
As a best practice, we avoid documenting future features; in case of unforeseen delays, having this information in the docs can cause liability issues. We'll be happy to add these in when they get added to the Preview!

Author: Hmm. I understand but we also want future customer to know what we are working on to improve this product.
Is there a way here ?

Reviewer: Yeah, I understand; I think it just doesn't fit very well into the docs, unfortunately. I've seen that sort of stuff in Preview pages and communicated directly with customers, and I think that's fine; it's just that our aim as a team is to try to make the docs as source of truth for current behaviour, and making promises about the future can muddy that. Sorry it's not a better answer for you!

Author: is this a blocker or is that ok to ship like this ? keeping in mind that we should ship these features in the next 3-4months

Reviewer: Sorry about my late reply; I was on PTO! Yeah, that's quite a long timeframe, and release plans can change quite a bit in that time, so we don't want to make any promises to be on the safe side. No worries, I just created https://github.com/DataDog/documentation/pull/37986 to take care of it, so you don't have to do anything else. 🙂
